lib/proof/domain/entities/gist_proof_entity.dart

```diff
@@ -9,6 +9,13 @@ class GistProofEntity {
   @override
   String toString() => "[GistProofEntity] {root: $root, proof: $proof}";
 
+  @override
+  Map<String, dynamic> toJson() => {
+        'root': root,
+        'proof': proof.toJson(),
+      }..removeWhere(
+          (dynamic key, dynamic value) => key == null || value == null);
```

Comment:
I saw you're removing `null` at every `toJson`, don't we need the value even if it's null in some case?